File: backend/internal/response/problems.go
package response

import "time"

type ProblemSummary struct {
	ProblemId       string    `json:"problem_id"`
	Title           *string   `json:"title"`
	Status          string    `json:"status"`
	NeedsReviewFlag bool      `json:"needs_review_flag"`
	CreatedAt       time.Time `json:"created_at"`
}

type GetProblemsResponse struct {
	Result []ProblemSummary `json:"result"`
	Total  int              `json:"total"`
}

type ProblemDetail struct {
	ProblemId       string    `json:"problem_id"`
	RawProblem      string    `json:"raw_problem"`
	Title           *string   `json:"title"`
	ProblemText     *string   `json:"problem_text"`
	AlgorithmTag    *string   `json:"algorithm_tag"`
	Difficulty      *string   `json:"difficulty"`
	AiHelp          *string   `json:"ai_help"`
	NeedsReviewFlag bool      `json:"needs_review_flag"`
	CreatedAt       time.Time `json:"created_at"`
	UpdatedAt       time.Time `json:"updated_at"`
}

type TodaysProblem struct {
	ProblemId       string    `json:"problem_id"`
	RawProblem      string    `json:"raw_problem"`
	Title           *string   `json:"title"`
	ProblemText     *string   `json:"problem_text"`
	AlgorithmTag    *string   `json:"algorithm_tag"`
	Difficulty      *string   `json:"difficulty"`
	Status          string    `json:"status"`
	AiHelp          *string   `json:"ai_help"`
	NeedsReviewFlag bool      `json:"needs_review_flag"`
	CreatedAt       time.Time `json:"created_at"`
	UpdatedAt       time.Time `json:"updated_at"`
}

type AIHelp struct {
	Concept     string `json:"concept"`
	Nudge       string `json:"nudge"`
	Approach    string `json:"approach"`
	Walkthrough string `json:"walkthrough"`
}

File: frontend/src/types/Problems.ts
export const Status = {
  Open: "Open",
  Failed: "Failed",
  Solved: "Solved",
  All: "All"
} as const

export type StatusType = typeof Status[keyof typeof Status]

export interface ProblemHistory {
  problem_id: string,
  title: string | null,
  status: StatusType,
  needs_review_flag: boolean,
  created_at: string
}

export interface ProblemResponse {
  result: ProblemHistory[],
  total: number
}

export interface Problem {
  problem_id: string,
  raw_problem: string,
  title: string | null,
  problem_text: string,
  algorithm_tag: string,
  difficulty: string,
  status: string,
  ai_help: string,
  needs_review_flag: boolean,
  created_at: string,
  updated_at: string
}

export interface AiHelp {
  concept: string,
  nudge: string,
  approach: string,
  walkthrough: string
}
